src/Select.js

```diff
@@ -886,8 +900,13 @@ export default class Select extends Component<Props, State> {
                 'aria-label': label,
                 role: 'group',
               }}
-              Heading={GroupHeading}
+              components={{ Heading: GroupHeading }}
               getStyles={this.getStyles}
+              menuOptions={menuOptions}
+              isOptionSelected={this.isOptionSelected}
+              selectValue={selectValue}
+              selectValues={this.selectValues}
+              group={group}
               {...group}
```

Comment:
I need to pass these 4 props to make the `Categorized` component work...
Not sure if this fits well in the overall architecture of the code.

I don't need to pass `group` though since I am already spreading it's attributes, I will remove that right now.

Reply:
actually, I need the `key` prop inside the `group` object but if I remove `group={group}` and rely on the attributes assigned by `{...group}` but even though I change [this line](https://github.com/JedWatson/react-select/pull/2341/files#diff-cdaff1a639e18d02379ab62c90551f29R56) to 
```
const { key, menuOptions: { render } } = this.props;
```

it breaks the code... I will need to investigate a bit..